test-loop-tests/src/utils/resharding.rs

```diff
@@ -1064,3 +1052,205 @@ pub(crate) fn promise_yield_repro_missing_trie_value(
+                    }
+                }
+                // GC happened a few blocks in the past.
+                (Some(resharding), latest)
+                    if latest == resharding + gc_num_epochs * epoch_length + 10 =>
+                {
+                    done.set(true)
+                }
+                // Catch-all case, do nothing.
+                (Some(_resharding), _latest) => {}
+                // Resharding didn't happen yet.
+                (None, _) => {
+                    let epoch_manager = client_actor.client.epoch_manager.as_ref();
+                    // Check if resharding will happen in this block.
+                    if next_block_has_new_shard_layout(epoch_manager, &tip) {
+                        tracing::debug!(target: "test", height=tip.height, "resharding height set");
+                        resharding_height.set(Some(tip.height));
+                        return;
+                    }
+                    // Before resharding, send transactions to trigger delayed receipts buffering,
+                    // just once.
+                    let will_reshard =
+                        epoch_manager.will_shard_layout_change(&tip.prev_block_hash).unwrap();
+                    if !will_reshard {
+                        return;
+                    }
+                    let epoch_length = client_actor.client.config.epoch_length;
+                    let epoch_start =
+                        epoch_manager.get_epoch_start_height(&tip.last_block_hash).unwrap();
+                    if tip.height + 5 < epoch_start + epoch_length {
+                        return;
+                    }
+                    burn_gas(
+                        &left_child_account,
+                        &right_child_account,
+                        &tip,
+                        Some(&pre_resharding_tx_sent),
+                    );
+                }
+            }
+        },
+    );
+    LoopAction::new(action_fn, succeeded)
+}
+
+/// Returns a tuple with the shard uids of: parent, left child, right child.
+fn get_resharded_shard_uids(
+    left_child_account: &AccountId,
+    right_child_account: &AccountId,
+    shard_layout_after_resharding: &ShardLayout,
+) -> (ShardUId, ShardUId, ShardUId) {
+    let left_child_shard_uid =
+        shard_layout_after_resharding.account_id_to_shard_uid(&left_child_account);
+    let right_child_shard_uid =
+        shard_layout_after_resharding.account_id_to_shard_uid(&right_child_account);
+    let parent_shard_uid = ShardUId::new(
+        3,
+        shard_layout_after_resharding.get_parent_shard_id(left_child_shard_uid.shard_id()).unwrap(),
+    );
+    (parent_shard_uid, left_child_shard_uid, right_child_shard_uid)
+}
+
+// Helper function to retrieve an indices-like value from the trie. This bypasses any other
+// intermediate layer (caching, memtrie, flat-storage).
+fn get_indices_value_for_shard<I: borsh::BorshDeserialize + Default>(
```

Comment:
nit: since it is in utils, we could rename it to be more general - to retrieve anything from the trie while bypassing intermediate layers.